expr/functions/timeShiftByMetric/function.go

```diff
@@ -0,0 +1,244 @@
+package timeShiftByMetric
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"regexp"
+	"strings"
+
+	"github.com/ansel1/merry"
+
+	"github.com/go-graphite/carbonapi/expr/helper"
+	"github.com/go-graphite/carbonapi/expr/interfaces"
+	"github.com/go-graphite/carbonapi/expr/types"
+	"github.com/go-graphite/carbonapi/pkg/parser"
+)
+
+type offsetByVersion map[string]int64
+
+type timeShiftByMetric struct {
+	interfaces.FunctionBase
+}
+
+func GetOrder() interfaces.Order {
+	return interfaces.Any
+}
+
+func New(configFile string) []interfaces.FunctionMetadata {
+	return []interfaces.FunctionMetadata{interfaces.FunctionMetadata{
+		F:    &timeShiftByMetric{},
+		Name: "timeShiftByMetric",
+	}}
+}
+
+func (f *timeShiftByMetric) Description() map[string]types.FunctionDescription {
```

Comment:
Just note, usually, the Description is at the end of a file. I don't insist, it may stay here to be consistent with Avito fork.

Reply:
Fixed